// Copyright 2017 Intel Corporation
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.
// -----------------------------------------------------------------------------

syntax = "proto3";

option java_multiple_files = true;
option java_package = "sawtooth.supplychain.protobuf";

// Agent State Representation
message Agent {
    string identifier = 1; // the hex encoded public key of the agent
    string name = 2; // the agent name
}

// Agent Container for storage in global state
message AgentContainer {
    // List of agents entries - more than one implies a state key collision
    repeated Agent entries = 1;
}

